ui/src/message_popup/author/question_page.jsx

```diff
@@ -0,0 +1,284 @@
+    }
+  },
+
+  removeStudent(studentId){
+    return function(){
+      this.setState({students: _.remove(_.clone(this.state.students), student => student.id !== studentId)});
+    }.bind(this);
+  },
+
+  onGoodExamplesChange(event, text){
+    this.setState({goodExamplesText: text});
+    this.getExamples();
+  },
+
+  onBadExamplesChange(event, text){
+    this.setState({badExamplesText: text});
+    this.getExamples();
+  },
+
+  getExamples(){
+    const {goodExamplesText, badExamplesText} = this.state;
+    var goodExamples = goodExamplesText.split('\n\n');
+    var badExamples = badExamplesText.split('\n\n');
+    goodExamples = goodExamples.map(example => example.trim());
+    badExamples = badExamples.map(example => example.trim());
+    _.remove(goodExamples, example => example == '');
+    _.remove(badExamples, example => example == '');
+    return {goodExamples, badExamples};
+  },
+
+  onChangeLearningObjective(event, index, value){
+    this.setState({learningObjective: _.find(learningObjectives, learningObjective => learningObjective.id === value)});
+  },
+
+  onChangeIndicator(event, value){
+    this.setState({indicator: _.find(indicators, indicator => indicator.id.toString() === value)});
+  },
+
+  saveCheckPassed(){
+    if(this.state.questionText !== '') return true;
+    return false;
+  },
+
+  onSaveEdit(){
+    console.log("");
+    console.log("Saving the following new question:");
+    const studentIds = this.state.students.map(student => student.id);
+    const text = this.state.questionText;
+    const {goodExamples, badExamples} = this.getExamples();
+    const id = _.maxBy(allQuestions, question => question.id).id + 1;
+    const question = {
+      studentIds,
+      id,
+      text,
+      examples: goodExamples,
+      nonExamples: badExamples,
+      indicatorId: this.state.indicator.id,
+      indicator: this.state.indicator,
+      learningObjectiveId: this.state.learningObjective.id,
+      learningObjective: this.state.learningObjective
+    };
+    console.log(question);
+    console.log("Also deleting and archiving the following question:");
+    console.log(this.props.originalQuestion);
+    console.log("");
+    this.returnToQuestions();
+  },
+
+  onCreate(){
+    console.log("");
+    console.log("Creating the following new question:");
+    const studentIds = this.state.students.map(student => student.id);
+    const text = this.state.questionText;
+    const {goodExamples, badExamples} = this.getExamples();
+    const id = _.maxBy(allQuestions, question => question.id).id + 1;
+    const question = {
+      studentIds,
+      id,
+      text,
+      examples: goodExamples,
+      nonExamples: badExamples,
+      indicatorId: this.state.indicator.id,
+      indicator: this.state.indicator,
+      learningObjectiveId: this.state.learningObjective.id,
+      learningObjective: this.state.learningObjective
+    };
+    console.log(question);
+    this.returnToQuestions();
```

Comment:
These look good!  For `onCreated`, `onDeleted` and `onUpdated`, we can update those in the next PR.